pkg/appfile/parser.go

```diff
@@ -130,7 +130,7 @@ func (p *Parser) GenerateAppFile(ctx context.Context, app *v1beta1.Application)
 	}
 
 	appfile.WorkflowMode = common.WorkflowModeDAG
-	if wfSpec := app.Spec.Workflow; wfSpec != nil {
+	if wfSpec := app.Spec.Workflow; wfSpec != nil && len(wfSpec.Steps) > 0 {
```

Comment:
What will happen without this change?

Reply:
Declaration like [this](https://github.com/oam-dev/kubevela/pull/2886/files#diff-dfc3f6e3b7231b7b41cf82b8e561f3cbf8191d214886e21d798f287f693806dbL450) will be `StepByStep` mode in workflow.